src/toast.tsx

```diff
@@ -172,6 +178,22 @@ const Toast: FC<ToastProps> = (props) => {
     };
   }, []);
 
+  /**
+   * On every update, changing the updateId will reset the animation
+   */
+  useEffect(() => {
+    if (updateId && closeTimeoutRef.current) {
+      clearTimeout(closeTimeoutRef.current);
+      closeTimeoutRef.current = setTimeout(() => {
+        handleClose();
+      }, duration);
+    }
+
+    return () => {
+      closeTimeoutRef.current && clearTimeout(closeTimeoutRef.current);
+    };
+  }, [updateId]);
```

Comment:
This array is not containing the other props to match the other `useEffect` already implement to set the duration of the animation. It should be fixed though, since it's not correct to skip props.

Reply:
Finally! I was waiting for this fix to come. Thanks for this @OctaBruzzesi !